pkg/pushsync/pushsync_test.go

```diff
@@ -2,4 +2,336 @@
+		{
+			Address: swarm.MustParseHexAddress("8000000000000000000000000000000000000000000000000000000000000000"), // binary 1000 -> po 0
+		},
+		{
+			Address: swarm.MustParseHexAddress("4000000000000000000000000000000000000000000000000000000000000000"), // binary 0100 -> po 1
+		},
+		{
+			Address: swarm.MustParseHexAddress("6000000000000000000000000000000000000000000000000000000000000000"), // binary 0110 -> po 1 want this one
+		},
+	}
+	closestPeer := connectedPeers[2].Address
+
+	// mock a connectivity between the nodes
+	p2ps := p2pmock.New(p2pmock.WithConnectFunc(func(ctx context.Context, addr ma.Multiaddr) (swarm.Address, error) {
+		return pivotNode, nil
+	}), p2pmock.WithPeersFunc(func() []p2p.Peer {
+		return connectedPeers
+	}))
+
+	// Create a full connectivity driver
+	fullDriver := full.New(mock.NewDiscovery(), addressbook.New(mockstate.NewStateStore()), p2ps, logger, pivotNode)
+	storer, err := localstore.New("", pivotNode.Bytes(), nil, logger)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	targetCalled := false
+	var mtx sync.Mutex
+
+	// setup the stream recorder to record stream data
+	recorder := streamtest.New(
+		streamtest.WithMiddlewares(func(f p2p.HandlerFunc) p2p.HandlerFunc {
+
+			// this is a custom middleware that is needed because of the design of
+			// the recorder. since we want to test only one unit, but the first message
+			// is supposedly coming from another node, we don't want to execute the handler
+			// when the peer address is the peer of `closestPeer`, since this will create an
+			// unnecessary entry in the recorder
+			return func(ctx context.Context, p p2p.Peer, s p2p.Stream) error {
+				if p.Address.Equal(connectedPeers[2].Address) {
+					mtx.Lock()
+					defer mtx.Unlock()
+					if targetCalled {
+						t.Fatal("target called more than once")
+					}
+					targetCalled = true
+					return nil
+				}
+				return f(ctx, p, s)
+			}
+		}),
+	)
+
+	ps := pushsync.New(pushsync.Options{
+		Streamer:      recorder,
+		Logger:        logger,
+		ClosestPeerer: fullDriver,
+		Storer:        storer,
+	})
+	defer ps.Close()
+
+	// TODO: this needs to go away once we have a simpler recorder
+	recorder.SetProtocols(ps.Protocol())
+
+	stream, err := recorder.NewStream(context.Background(), pivotNode, nil, pushsync.ProtocolName, pushsync.ProtocolVersion, pushsync.StreamName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer stream.Close()
+	w, _ := protobuf.NewWriterAndReader(stream)
+
+	// this triggers the handler of the pivot with a delivery stream
+	err = w.WriteMsg(&pb.Delivery{
+		Address: chunkAddress.Bytes(),
+		Data:    chunkData,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for i := 0; i < 20; i++ {
+		mtx.Lock()
+		called := targetCalled
+		mtx.Unlock()
+
+		if !called {
+			time.Sleep(10 * time.Millisecond)
+		}
+	}
+
+	records, err := recorder.Records(closestPeer, pushsync.ProtocolName, pushsync.ProtocolVersion, pushsync.StreamName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if l := len(records); l != 1 {
```

Comment:
why not use the same construction as in the previous test here for enforcing only one record?